Correct min/max temperature parsing

mmore500 opened the pull request:
The previous scheme computed the average over maximum expected
temperatures throughout a day and minimum expected temperatures
throughout a day.
(Essentially, under the previous scheme minimum and maximum temperatures
were daily averages that accounted for uncertainty in the forecast).
Now, minimum temperature is computed as the minimum of
forecast temperatures for a day and the maximum temperature is
computed as the maximum of forecast temperatures for a day.

zagortenay333 commented:
The thing is that I don't know how their API actually works. My assumption was/is that the 3-hourly records they return are referring to the whole day. I.e., we get multiple approximations for that one day, so I averaged it out.

Is any particular temp_min referring to the 3-hours or the whole day?

mmore500 commented:
from my weather.json:

"temp":38.91,"temp_min":38.91,"temp_max":39.28
"dt_txt":"2017-11-27 03:00:00"

...

"temp":49.33,"temp_min":49.33,"temp_max":49.33
"dt_txt":"2017-12-01 00:00:00"

"temp":44.92,"temp_min":44.92,"temp_max":44.92
"dt_txt":"2017-12-01 03:00:00"},

"temp":40.38,"temp_min":40.38,"temp_max":40.38
"dt_txt":"2017-12-01 06:00:00"

"temp":39.23,"temp_min":39.23,"temp_max":39.23
"dt_txt":"2017-12-01 09:00:00"

"temp":35.7,"temp_min":35.7,"temp_max":35.7
"dt_txt":"2017-12-01 12:00:00"

"temp":37.55,"temp_min":37.55,"temp_max":37.55
"dt_txt":"2017-12-01 15:00:00"}

mmore500 commented:
I haven't looked at all at their API specifications, but empirically from my `weather.json` it looks like min/max are some kind of confidence interval around temperature approximation at a particular point in time. Most (but not all) of the time, min/max and plain temperature all have the same value and the min/max/plain temp all track warming and cooling throughout the day as one would expect.

zagortenay333 commented:
I was talking about the temp_min in the forecast.json file. I've no clue what the temp_min in weather.json is.

In the forecast file I don't know whether the temp_min is a forecast for the next 3 hours or an updated forecast for the whole day.

Asked here: https://openweathermap.desk.com/customer/en/portal/questions/17196596-5-day-forecast?new=17196596

mmore500 commented:
Sorry, in my above statements I misspoke. I meant to refer to `forecast.json` but accidentally (and incorrectly) referred to `weather.json`. Those snippets are from `forecast.json`. Looking at those snippets, it seems like `temp_min` is a forecast for the next 3 hours. Hopefully someone at openweathermap will be able to confirm that for you.

mmore500 commented:
Unrelated, I added another commit that cleans out unused variables and debugging logging statements I had added in from the new functions `get_min` and `get_max` in the script `parse_weather`.

mmore500 commented:
Looks like it's the forecast for the next 3 hours.
[https://openweathermap.desk.com/customer/en/portal/questions/17196596-5-day-forecast?new=17196596](https://openweathermap.desk.com/customer/en/portal/questions/17196596-5-day-forecast?new=17196596)

Are we good to merge?
